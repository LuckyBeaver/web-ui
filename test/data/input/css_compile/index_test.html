<!DOCTYPE html>
<!--
Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <link rel="import" href="components/comp1.html">
  <link rel="import" href="components/comp2.html">
  <link rel="import" href="components/comp3.html">
  <link rel="import" href="components/comp1_reset.html">
  <link rel="import" href="components/comp2_reset.html">
  <title>CSS Compile-Time</title>
</head>
<body>
  <style>
    div {
      font-size:18pt;
      font-weight: bold;
      color: red;
      margin: 5px;
     }

    .a2 {
       display: inline;
       color: green;
       border: 1px solid green;
    }
  </style>

  <h1>Apply Author Styles</h1>
  <div>test #1 (red)</div>
  <div>
    <x-comp1>x-comp1 (blue)</x-comp1>
  </div>
  <div>
    <x-comp3>x-comp3 (green/green border)</x-comp3>
  </div>
  <div>
    <x-comp1-reset>x-comp1-reset (yellow/yellow border)</x-comp1-reset>
  </div>
  <div>test #4 (red)</div>

  <h1>NO Apply Author Styles</h1>
  <div class="a2">Test with class a2 (green solid border display: inline)</div>
  <div>
    <x-comp2>x-comp2 with class a2 (purple solid border display: inline)</x-comp2>
  </div>
  <div>
    <x-comp2-reset>x-comp2-reset with class a2 (orange dashed border display: block;)</x-comp2-reset>
  </div>
  <div class="a2">Test with class a2 (green solid border display: inline)</div>

  <script type="application/dart">
import 'dart:async';
import 'dart:html';
import 'package:unittest/unittest.dart';
import 'package:web_ui/web_ui.dart';

main() {
  useShadowDom = false;

  Timer.run(() {
    var divs = queryAll('div');
    expect(divs.length, 14);

    var items = queryAll('div .a2');
    expect(items.length, 3);

    var xComp1 = queryAll('[is="x-comp1"]');
    expect(xComp1.length, 1);

    var xComp1Reset = queryAll('[is="x-comp1-reset"]');
    expect(xComp1Reset.length, 1);

    var xComp2 = queryAll('[is="x-comp2"]');
    expect(xComp2.length, 1);

    var xComp2Reset = queryAll('[is="x-comp2-reset"]');
    expect(xComp2Reset.length, 1);

    window.postMessage('done', '*');
   });
}
  </script>
</body>
</html>
