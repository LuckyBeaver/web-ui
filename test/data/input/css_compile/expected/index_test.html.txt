Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x480
  RenderBlock {HTML} at (0,0) size 800x480
    RenderBody {BODY} at (8,21) size 784x451
      RenderBlock {H1} at (0,0) size 784x37
        RenderText {#text} at (0,0) size 278x36
          text run at (0,0) width 278: "Apply Author Styles"
      RenderBlock {DIV} at (5,58) size 774x28 [color=#FF0000]
        RenderText {#text} at (0,0) size 123x26
          text run at (0,0) width 123: "test #1 (red)"
      RenderBlock {DIV} at (5,90) size 774x38 [color=#FF0000]
        RenderBlock (anonymous) at (0,0) size 774x0
          RenderInline {SPAN} at (0,0) size 0x0
            RenderText {#text} at (0,0) size 0x0
        RenderBlock (anonymous) at (0,0) size 774x37
          RenderBlock {DIV} at (5,0) size 764x37 [color=#0000FF]
            RenderText {#text} at (5,5) size 153x26
              text run at (5,5) width 153: "x-comp1 (blue)"
            RenderText {#text} at (0,0) size 0x0
        RenderBlock (anonymous) at (0,42) size 774x0
          RenderInline {SPAN} at (0,0) size 0x0
            RenderText {#text} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
      RenderBlock {DIV} at (5,132) size 774x28 [color=#FF0000]
        RenderInline {SPAN} at (0,0) size 321x26
          RenderText {#text} at (0,0) size 0x0
          RenderInline {DIV} at (0,0) size 311x28 [color=#008000] [border: (1px solid #008000)]
            RenderText {#text} at (0,0) size 0x0
            RenderText {#text} at (6,0) size 309x26
              text run at (6,0) width 309: "x-comp3 (green/green border)"
            RenderText {#text} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
        RenderText {#text} at (0,0) size 0x0
      RenderBlock {DIV} at (5,164) size 774x30 [color=#FF0000]
        RenderBlock (anonymous) at (0,0) size 774x0
          RenderInline {SPAN} at (0,0) size 0x0
            RenderText {#text} at (0,0) size 0x0
        RenderBlock (anonymous) at (0,0) size 774x29
          RenderBlock {DIV} at (0,0) size 774x29 [color=#FFFF00] [border: (1px solid #FFFF00)]
            RenderText {#text} at (1,1) size 383x26
              text run at (1,1) width 383: "x-comp1-reset (yellow/yellow border)"
            RenderText {#text} at (0,0) size 0x0
        RenderBlock (anonymous) at (0,29) size 774x0
          RenderInline {SPAN} at (0,0) size 0x0
            RenderText {#text} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
      RenderBlock {DIV} at (5,198) size 774x28 [color=#FF0000]
        RenderText {#text} at (0,0) size 123x26
          text run at (0,0) width 123: "test #4 (red)"
      RenderBlock {H1} at (0,246) size 784x38
        RenderText {#text} at (0,0) size 334x36
          text run at (0,0) width 334: "NO Apply Author Styles"
      RenderBlock (anonymous) at (0,305) size 784x28
        RenderInline {DIV} at (0,0) size 542x28 [color=#008000] [border: (1px solid #008000)]
          RenderText {#text} at (6,0) size 540x26
            text run at (6,0) width 540: "Test with class a2 (green solid border display: inline)"
        RenderText {#text} at (0,0) size 0x0
      RenderBlock {DIV} at (5,337) size 774x28 [color=#FF0000]
        RenderInline {SPAN} at (0,0) size 636x26
          RenderText {#text} at (0,0) size 0x0
          RenderInline {DIV} at (0,0) size 626x58 [color=#800080] [border: (1px solid #800080)]
            RenderText {#text} at (0,0) size 0x0
            RenderText {#text} at (21,0) size 594x26
              text run at (21,0) width 594: "x-comp2 with class a2 (purple solid border display: inline)"
            RenderText {#text} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
        RenderText {#text} at (0,0) size 0x0
      RenderBlock {DIV} at (5,379) size 774x30 [color=#FF0000]
        RenderBlock (anonymous) at (0,0) size 774x0
          RenderInline {SPAN} at (0,0) size 0x0
            RenderText {#text} at (0,0) size 0x0
        RenderBlock (anonymous) at (0,0) size 774x29
          RenderBlock {DIV} at (15,0) size 744x29 [color=#FFA500] [border: (1px dashed #FFA500)]
            RenderText {#text} at (1,1) size 684x26
              text run at (1,1) width 684: "x-comp2-reset with class a2 (orange dashed border display: block;)"
            RenderText {#text} at (0,0) size 0x0
        RenderBlock (anonymous) at (0,44) size 774x0
          RenderInline {SPAN} at (0,0) size 0x0
            RenderText {#text} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
      RenderBlock (anonymous) at (0,423) size 784x28
        RenderInline {DIV} at (0,0) size 542x28 [color=#008000] [border: (1px solid #008000)]
          RenderText {#text} at (6,0) size 540x26
            text run at (6,0) width 540: "Test with class a2 (green solid border display: inline)"
        RenderText {#text} at (0,0) size 0x0
        RenderText {#text} at (0,0) size 0x0
